### PR TITLE
Feature tests for workflow update

### PR DESCRIPTION
**Pull request by mmcshane**
## What was changed

Add feature tests for workflow update

## Why?

Check that e2e integration and some edge cases are functional for workflow update feature.

## Checklist

1. Closes 

2. How was this tested:

3. Any docs updates needed?

### CODE REVIEW

**Comment by mmcshane** [comment]
FYI these have been stable for a while now but needed the test harness to be able to skip tests without erroring out.

File: features/update/activities/feature.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+			updateActivity,
+		)
+		runner.Require.NoError(err)
+
+		var result int
+		runner.Require.NoError(handle.Get(ctx, &result))
+		runner.Require.Equal(activityResult*activityCount, result)
+
+		runner.Require.NoError(runner.Client.SignalWorkflow(ctx, run.GetID(), run.GetRunID(), shutdownSignal, nil))
+		updateutil.RequireNoUpdateRejectedEvents(ctx, runner)
```
Comment (Quinn-With-Two-Ns) [inline features/update/activities/feature.go:46]: Do we need this check here since `UpdateHandlerOptions` doesn't have a validator? Or is the intention to have this check for every sync. update test? I notice it was missing from the panic test case?

File: features/update/validation_replay/README.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# Replay does not perform update validation
```
Comment (Quinn-With-Two-Ns) [inline features/update/validation_replay/README.md:1]: Should we also test the update function does run on replay?

File: features/update/non_durable_reject/README.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
+# Update Rejections Do Not Take Up Space In History
+
+When updates are registered with a validation handler and that validation
+handler rejects an update, neither the invocation of the update nor the
```
Comment (Quinn-With-Two-Ns) [inline features/update/non_durable_reject/README.md:4]: > nor the rejection nor the rejection

File: features/update/self/README.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
+# Workflow sends an update to itself
+
+Workflows can send updates back to themselves
+
+# Detailed spec
+
+A workflow can use the workflowservice client in an activity to send an update
```
Comment (Quinn-With-Two-Ns) [inline features/update/self/README.md:7]: More a question on updates use case: In this test if there was some failure your activity to send a update could be called more then one correct? Could update validators be used to dedup the update if a UUID was attached?

**Review by Quinn-With-Two-Ns — COMMENTED** [review]
Do we ever test am update returning in error in the update handler function?

File: features/update/activities/feature.go (hunk 1 above)
Comment (mmcshane) [reply features/update/activities/feature.go:46]: I think my intent was to add it to all of them. That it's missing from the panic test is just an oversight - I'll add it.

File: features/update/validation_replay/README.md (hunk 2 above)
Comment (mmcshane) [reply features/update/validation_replay/README.md:1]: Maybe I'm wrong about what the harness does during replay but if the update didn't run under replay then we wouldn't see the activity and I would have expected that to get detected as mismatched history?

File: features/update/non_durable_reject/README.md (hunk 3 above)
Comment (mmcshane) [reply features/update/non_durable_reject/README.md:4]: :+1:

File: features/update/self/README.md (hunk 4 above)
Comment (mmcshane) [reply features/update/self/README.md:7]: We made the choice not to include the UpdateID in the invocation of the handler but the user is certainly free to include an idempotency token in their application-layer update argument. The server can and does dedup by update ID in that a second request with the same update ID will wait on the same update protocol instance in the history service. You can think of this as finding and waiting on the same Future-ish Get()

File: features/update/validation_replay/README.md (hunk 2 above)
Comment (Quinn-With-Two-Ns) [reply features/update/validation_replay/README.md:1]: No your right ignore me. Yeah the history should fail determinism if the activity didn't run

**Review by Quinn-With-Two-Ns — APPROVED** [review]
LGTM, though I am not sure how much coverage we want with these tests? 

Two tests we don't have now that I think would be useful are testing updates with queries and testing update handlers that return an error.
